### PR TITLE
Use testing.T.TempDir() instead of os.TempDir() 

### LINKED ISSUE #147: fix unit tests to use testing.T.TempDir instead of os.TempDir
let's use https://pkg.go.dev/testing#T.TempDir for our unit tests so that it cleans up when the testing is done

### PR DESCRIPTION
**Pull request by threebithacker**
... to have `go test` clean up after itself automatically.

Closes #147

Let's see what CI thinks about this. **DON'T MERGE YET**

### CODE REVIEW

**Comment by threebithacker** [comment]
Looks like the reunion test is leaving threads running after the test completes. Will look into that.

**Review by david415 — APPROVED** [review]

**Review by david415 — APPROVED** [review]

**Comment by mixmasala** [comment]
doesn't seem to have exited cleanly yet; 
```
panic: test timed out after 30m0s

goroutine 11138 [running]:
testing.(*M).startAlarm.func1()
	/opt/hostedtoolcache/go/1.18.10/x64/src/testing/testing.go:2029 +0xbb
created by time.goFunc
	/opt/hostedtoolcache/go/1.18.10/x64/src/time/sleep.go:176 +0x48

goroutine 1 [chan receive, 29 minutes]:
testing.tRunner.func1()
	/opt/hostedtoolcache/go/1.18.10/x64/src/testing/testing.go:1405 +0x865
testing.tRunner(0xc00008dba0, 0xc0000cbb68)
	/opt/hostedtoolcache/go/1.18.10/x64/src/testing/testing.go:1445 +0x255
testing.runTests(0xc00009d2c0?, {0x9c3140, 0x5, 0x5}, {0x2?, 0x7ff3a3a55138?, 0xa137c0?})
	/opt/hostedtoolcache/go/1.18.10/x64/src/testing/testing.go:1837 +0x7e5
testing.(*M).Run(0xc00009d2c0)
	/opt/hostedtoolcache/go/1.18.10/x64/src/testing/testing.go:1719 +0xa72
main.main()
	_testmain.go:105 +0x3aa

goroutine 6 [chan receive]:
github.com/awnumar/memguard/core.NewCoffer.func1(0x0?)
	/root/go/pkg/mod/github.com/awnumar/memguard@v0.22.3/core/coffer.go:39 +0x6a
created by github.com/awnumar/memguard/core.NewCoffer
	/root/go/pkg/mod/github.com/awnumar/memguard@v0.22.3/core/coffer.go:36 +0x1dd

goroutine 1019 [chan send, 29 minutes]:
github.com/katzenpost/katzenpost/reunion/client.(*Exchange).sentUpdateOK(0xc000118000)
	/home/runner/work/katzenpost/katzenpost/reunion/client/exchange.go:268 +0x13b
github.com/katzenpost/katzenpost/reunion/client.TestClientServerBasics1(0xc000a4d380)
	/home/runner/work/katzenpost/katzenpost/reunion/client/exchange_test.go:185 +0xe05
testing.tRunner(0xc000a4d380, 0x8039b8)
	/opt/hostedtoolcache/go/1.18.10/x64/src/testing/testing.go:1439 +0x214
created by testing.(*T).Run
	/opt/hostedtoolcache/go/1.18.10/x64/src/testing/testing.go:1486 +0x725

goroutine 1021 [select]:
github.com/katzenpost/katzenpost/reunion/server.(*Server).worker(0xc00017c000)
	/home/runner/work/katzenpost/katzenpost/reunion/server/server.go:289 +0x3ef
github.com/katzenpost/katzenpost/core/worker.(*Worker).Go.func1()
	/home/runner/work/katzenpost/katzenpost/core/worker/worker.go:39 +0x74
created by github.com/katzenpost/katzenpost/core/worker.(*Worker).Go
	/home/runner/work/katzenpost/katzenpost/core/worker/worker.go:37 +0x10f
FAIL	github.com/katzenpost/katzenpost/reunion/client	1800.037s
-test.shuffle 1676213518787115783
```

**Comment by threebithacker** [comment]
No, now the `.Run()` run forever instead of running after the test concludes. Slight improvement, but not great either.

**Comment by mixmasala** [comment]
@threebithacker: If you open the PR from your own GH I can't push any changes. See:
https://github.com/katzenpost/katzenpost/tree/testing_cleanup_tempdir

**Comment by mixmasala** [comment]
> @threebithacker: If you open the PR from your own GH I can't push any changes. See: https://github.com/katzenpost/katzenpost/tree/testing_cleanup_tempdir

ok apparently I can edit the file here though

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -10,19 +10,20 @@ jobs:
...
-        uses: actions/checkout@v3
+          cache: true
+          check-latest: true
 
       - name: Generate mixnet configuration
         run: cd genconfig && go build && ./genconfig -v -b /conf -o ../docker/voting_mixnet/
 
       - name: Run all package unit tests
-        run: sudo sh -c "ulimit -l 64435 && CGO_CFLAGS_ALLOW=-DPARAMS=sphincs-shake-256f GORACE=history_size=7 go test -coverprofile=coverage.out -race -v -timeout 0 ./..."
+        run: sudo sh -c "ulimit -l 64435 && CGO_CFLAGS_ALLOW=-DPARAMS=sphincs-shake-256f GORACE=history_size=7 go test -shuffle on -failfast -coverprofile=coverage.out -race -v -timeout 30m ./..."
```
Comment (mixmasala) [inline .github/workflows/go.yml:26]: good, though I'd like to know why this test sometimes hangs forever and fix the underlying issue.

File: reunion/client/exchange_test.go [hunk 2]
```diff
@@ -102,12 +109,14 @@ func TestClientServerBasics1(t *testing.T) {
 	go func() {
 		for {
 			update := <-aliceUpdateCh
+			require.NoError(update.Error)
 			if len(update.Result) > 0 {
 				aliceResult = update.Result
-				fmt.Printf("\n Alice got result: %s\n\n", update.Result)
-				wg.Done()
+				t.Log("Alice got result:", update.Result)
+				break
```
Comment (mixmasala) [inline reunion/client/exchange_test.go:116]: agree this worker doesn't exit but doesn't updateCh receive more than one message?

Comment (mixmasala) [reply reunion/client/exchange_test.go:116]: what can be done is to read from aliceUpdateCh like so:
update, ok := aliceUpdateCh
if !ok { // channel closed
  return
  }

Comment (mixmasala) [reply reunion/client/exchange_test.go:116]: (and then ensure that the channel is closed at teardowN)

File: reunion/client/exchange_test.go [hunk 3]
```diff
@@ -124,13 +133,15 @@ func TestClientServerBasics1(t *testing.T) {
...
 		for {
-			update := <-bobUpdateCh
+			update := <-bobUpdateCh // this never gets shut down
+			require.NoError(update.Error)
 			if len(update.Result) > 0 {
 				bobResult = update.Result
-				fmt.Printf("\n Bob got result: %s\n\n", update.Result)
-				wg.Done()
+				t.Log("Bob got result:", update.Result)
+				break
```
Comment (mixmasala) [inline reunion/client/exchange_test.go:141]: see above

**Review by mixmasala — COMMENTED** [review]
looks good, if the reunion tests are passing. I thought that the goroutines consuming reunion events needed to receive more than one message, but not quite sure.

**Pull request closed without merging**